test/unit/org/apache/cassandra/cql3/validation/operations/SelectSingleColumnRelationTest.java

```diff
@@ -63,6 +63,8 @@ public void testInvalidCollectionNonEQRelation() throws Throwable
                              "SELECT * FROM %s WHERE c = 0 AND b <= ?", set(0));
         assertInvalidMessage("Collection column 'b' (set<int>) cannot be restricted by a 'IN' relation",
                              "SELECT * FROM %s WHERE c = 0 AND b IN (?)", set(0));
+        assertInvalidMessage("Collection column 'b' (set<int>) cannot be restricted by a 'BETWEEN' relation",
```

Comment:
Shall we add some test case to `testClusteringColumnRelationsWithClusteringOrder` and `testWithUnsetValues` in this class?